verify.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -e
+
+function validate() {
+    python_version=$1
+
+    echo "Validating version $python_version..."
+
+    tag=py-ipfs-http-client-verify:$python_version
+
+    docker build --build-arg PYTHON_VERSION=$python_version -t $tag .
+    docker run -it -v $(pwd):/source:ro -w /usr/src/app $tag tox -e styleck -e typeck
+}
+
+if [ -z "$1" ]; then
+    echo "Validating supported versions..."
+
+    validate 3.5.10
+    validate 3.6.13
+    validate 3.7.10
+    validate 3.8.8
```

Comment:
Is is possible to use aliases like `3.5`, `3.6`, … here to avoid listing all these detailed versions?

Reply:
You can; it's a trade off between build consistency and being automatically on a later point release of Python.

Which would you prefer?

I think ideally it would use the same source of truth as the rest of the build and scrape the config files, but I didn't get that far.

Reply:
Consistency is probably better, you're right! It's probably best to test the supported *earliest* version of each Python minor release.

> I think ideally it would use the same source of truth as the rest of the build and scrape the config files, but I didn't get that far.

The current “single source of truth” would be the `tool.flit.metadata.requires-python` key in `pyproject.toml`, whose value would effectively evaluate to the following Python minimum minor version configuration:

```sh
    validate 3.5.4
    validate 3.6.2
    validate 3.7.2
    validate 3.8.0
```

It's not straightforward at all to convert these values programmatically through. :-/